src/ProjectFileManager.cpp

```diff
@@ -1304,161 +1304,84 @@ class ImportProgress final
-      {
-         clip->SetRawAudioTempo(syncInfo.rawAudioTempo);
-         clip->TrimQuarternotesFromRight(syncInfo.excessDurationInQuarternotes);
-         clip->StretchBy(syncInfo.stretchMinimizingPowOfTwo);
-         ph.ModifyState(true);
-      }
-      else
-      {
-         const auto ans = UserWantsMirResultToConfigureProject(
-            *pProj, syncInfo.rawAudioTempo);
-         if (isBeatsAndMeasures || ans.yes)
-         {
-            clip->SetRawAudioTempo(syncInfo.rawAudioTempo);
-            clip->TrimQuarternotesFromRight(
-               syncInfo.excessDurationInQuarternotes);
-         }
-         if (ans.yes)
-         {
-            AdornedRulerPanel::Get(*pProj).SetTimeDisplayMode(
-               TimeDisplayMode::BeatsAndMeasures);
-            auto& projTimeSignature = ProjectTimeSignature::Get(*pProj);
-            projTimeSignature.SetTempo(syncInfo.rawAudioTempo);
-            if (syncInfo.timeSignature.has_value())
-            {
-               projTimeSignature.SetUpperTimeSignature(
-                  MIR::GetNumerator(*syncInfo.timeSignature));
-               projTimeSignature.SetLowerTimeSignature(
-                  MIR::GetDenominator(*syncInfo.timeSignature));
-            }
-         }
-         else if (isBeatsAndMeasures)
-            clip->StretchBy(syncInfo.stretchMinimizingPowOfTwo);
-         if (!ans.readFromPreference && ans.yes)
-            ph.PushState(
-               XO("Configure Project from Music File"),
-               XO("Automatic Music Configuration"));
-         else
-            ph.ModifyState(true);
-      }
-   });
+   std::vector<std::shared_ptr<MIR::AnalyzedAudioClip>> analyzedClips;
+   analyzedClips.reserve(readers.size());
+   std::transform(
+      readers.begin(), readers.end(), std::back_inserter(analyzedClips),
+      [&](const std::shared_ptr<ClipMirAudioReader>& reader) {
+         MIR::ProjectSyncInfoInput input {
+            *reader,      reader->filename, reader->tags,       reportProgress,
+            projectTempo, projectWasEmpty,  isBeatsAndMeasures,
+         };
+         auto syncInfo = MIR::GetProjectSyncInfo(input);
+         ++count;
+         return std::make_shared<AnalyzedWaveClip>(reader, syncInfo);
+      });
+   return analyzedClips;
 }
 } // namespace
 
-bool ProjectFileManager::ImportOneOfOne(
-   const FilePath& fileName, bool addToHistory)
+bool ProjectFileManager::Import(
+   const std::vector<FilePath>& fileNames, bool addToHistory)
 {
-   return Import(fileName, 1, 1, addToHistory);
+   const auto projectWasEmpty =
+      TrackList::Get(mProject).Any<WaveTrack>().empty();
+   std::vector<std::shared_ptr<ClipMirAudioReader>> resultingReaders;
+   const auto success = std::all_of(
+      fileNames.begin(), fileNames.end(), [&](const FilePath& fileName) {
+         ClipMirAudioReader* pResultingReader = nullptr;
+         const auto success = Import(fileName, addToHistory, pResultingReader);
+         if (success && pResultingReader)
```

Comment:
Not sure if it's the case in the real world, but there is a code path where `ClipMirAudioReader` could be created but `Import` returns `false`. Could be easily fixed with `unique_ptr`

Reply:
It would be created but then destroyed when this lambda exits if `!success`. What is your concern?
I haven't managed moving a vector of unique pointers around without invoking the unique_ptr copy ctor, yielding an error.
A couple of lines below, `if (!resultingReaders.empty()` should probably be changed to `if (success && !resultingReaders.empty())`, though.

Reply:
My concern is that `pResultingReader` is a raw pointer type. There seem to be a case where `Import` creates a `ClipMirAudioReader`, but then returns false, and object leaks because it's not yet wrapped with any type of smart pointer. 
But you've already said that you'll change the type of the argument to `shared_ptr&` in `Import`, that should solve the problem